src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -1336,7 +1341,14 @@ module SimulationCode =
         let outType  = op.Signature.ReturnType   |> roslynTypeName context
 
         let constructors = [ (buildConstructor context name) ]
-        let properties = buildName name :: buildFullName context.current.Value :: buildOperationInfoProperty inType outType nonGenericName :: buildOpsProperties context opNames
+        let properties = 
+            let opProperties = buildOpsProperties context opNames
+            buildName name :: 
+            buildFullName context.current.Value :: 
+            if globalContext.entryPoints |> Seq.contains op.FullName then 
```

Comment:
A question to understand a bit better what is going on here. It seems like the EntryPointInfo property and the properties that are generated based on the targeted architecture are only going to be generated for Operations that are marked as entry points. However, I don't get how this is done. What is the purpose of "Seq.contains op.FullName"?

Reply:
The global context contains an ImmutableArray of entry points. We do currently enforce that no more than one entry point exists, but for the sake of keeping data structures flexible, the compilation actually contains an ImmutableArray. Seq.contains just checks whether that array lists op as entry point.